### PR TITLE
HW-54226. Extracted transfer method repository

### PR DESCRIPTION
**Pull request by azakrevska-epam**

### CODE REVIEW

File: transfermethodrepository/build.gradle [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+dependencies {
+    implementation project(":common")
+
+    androidTestImplementation "androidx.test:rules:$testRulesVersion"
+    testImplementation "org.robolectric:robolectric:$robolectricVersion"
+    testImplementation "com.squareup.okhttp3:mockwebserver:$mockWebServerVersion"
+}
+
+
+def aarFile = file("$buildDir/outputs/aar/userrepository-$version" + ".aar")
```
Comment (fmattos-hw) [inline transfermethodrepository/build.gradle]: I think it shouldn't be userrepository here. Maybe we can use the project name variable?
`def aarFile = file("$buildDir/outputs/aar/$project.name-$version" + ".aar")`

File: transfermethodrepository/build.gradle [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            url = isReleaseVersion ? releasesRepoUrl : snapshotsRepoUrl
+            credentials {
+                username sonatypeUsername
+                password sonatypePassword
+            }
+        }
+    }
+
+    publications {
+        hyperwalletReceiptUi(MavenPublication) {
```
Comment (fmattos-hw) [inline transfermethodrepository/build.gradle]: receipt?

File: transfermethodrepository/build.gradle [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+}
+
+tasks.withType(Sign) {
+    onlyIf {
+        isReleaseVersion && sonatypeUsername?.trim() && sonatypePassword?.trim()
+    }
+}
+
+signing {
+    sign publishing.publications.hyperwalletReceiptUi
```
Comment (fmattos-hw) [inline transfermethodrepository/build.gradle]: receipt?

File: transfermethodrepository/build.gradle [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+                username sonatypeUsername
+                password sonatypePassword
+            }
+        }
+    }
+
+    publications {
+        hyperwalletReceiptUi(MavenPublication) {
+            groupId = hyperwalletGroupId
+            artifactId = 'transfermethodrepository'
```
Comment (fmattos-hw) [inline transfermethodrepository/build.gradle]: we could use `artifactId = project.name`

File: ui/src/main/java/com/hyperwallet/android/ui/repository/RepositoryFactory.java [hunk 5]
```diff
@@ -27,6 +27,9 @@
 
 package com.hyperwallet.android.ui.repository;
 
+import com.hyperwallet.android.ui.transfermethod.repository.TransferMethodRepository;
```
Comment (fmattos-hw) [inline ui/src/main/java/com/hyperwallet/android/ui/repository/RepositoryFactory.java]: The class RepositoryFactory.java  should be deprecated. We need to create a factory class (TransferMethodRepositoryFactory) for the TransferMethodRepository in the transfermethodrepository module. Same for the other repositories 
@vshcherbyna-epam this is applicable to you too

File: transfermethodrepository/build.gradle (hunk 2 above)
Comment (azakrevska-epam) [reply transfermethodrepository/build.gradle]: my mistake, fixed

File: transfermethodrepository/build.gradle (hunk 3 above)
Comment (azakrevska-epam) [reply transfermethodrepository/build.gradle]: removed

File: ui/src/main/java/com/hyperwallet/android/ui/repository/RepositoryFactory.java (hunk 5 above)
Comment (azakrevska-epam) [reply ui/src/main/java/com/hyperwallet/android/ui/repository/RepositoryFactory.java]: Done

**Pull request merged**